docs/important/relevant_functionalities_de.md

```diff
@@ -0,0 +1,1250 @@
+
+```json
+{
+  ...,
+  "ServiceAccounts": [
+    {
+      "Type": "produzentendogu",
+      "Kind": "dogu"
+    }
+  ],
+  ...
+}
+```
+
+Bei einem Service Account handelt es sich um eine besondere Form der Abhängigkeit. Trotzdem ist es sinnvoll, das Produzenten-Dogu in seiner [Dependency-Liste]() zu führen. So wird in den unterschiedlichen Phasen einer Dogu-Installation sichergestellt, dass das Produzenten-Dogu wirklich zur Benutzung bereitsteht:
+
+
+```json
+{
+  ...,
+  "Dependencies": [
+    {
+      "name": "produzentendogu",
+      "type": "dogu"
+    }
+  ],
+  "ServiceAccounts": [
+    {
+      "Type": "produzentendogu",
+      "Kind": "dogu"
+    }
+  ],
+  ...
+}
+```
+
+## Dogu-Upgrades
+
+Das Upgrade eines Dogus kann mit `cesapp upgrade dogu <doguname>` durchgeführt werden. Dabei verwaltet die cesapp alle wichtigen Schritte, um das Dogu erfolgreich zu upgraden.
+
+Der Upgrade-Prozess enthält u.A. folgende Schritte:
+- Anzeigen von Upgrade Notifications, falls vorhanden
+- Download des neuen Dogu-Images
+- Sicherstellen, dass das Dogu healthy ist
+- Sicherstellen, dass alle abhängigen Dogus verfügbar sind
+- Ausführen des Pre-Upgrade-Skripts im alten Dogu, falls vorhanden
+- Ersetzen des alten Dogus durch das neue
+- Starten des neuen Dogus
+- Ausführen des Post-Upgrade-Skripts im neuen Dogu, falls vorhanden
+
+Falls es beim Dogu-Upgrade notwendig ist, vor- oder nachbereitende Schritte durchzuführen, können optionale Pre- und Post-Upgrade-Skripte erstellt werden.
+Als Beispiel kann man eine Software betrachten, die in einer neueren Version eine andere Struktur der Datenbank benötigt.
+In einem solchen Fall ist es möglich, Upgrade-Skripte zu schreiben, um eine Migration der Datenbank nach einem Dogu-Upgrade zu starten.
+
+Außerdem kann man über wichtige Änderungen beim Upgrade eines Dogus durch ein Upgrade-Notification-Skript hinweisen lassen.
+
+Alle im Folgenden beschriebenen Upgrade-Skripte sollten (neben bspw. dem `startup.sh`-Skript) im `resources`-Ordner abgelegt werden.
+
+### Ermittlung der Upgrade-Pfade
+
+Bei der Entwicklung der Dogus ist dafür zu sorgen, dass Upgrades eines Dogus auch über mehrere Versionen hinweg funktionieren. Dafür müssen die Upgrade-Skripte so gestaltet sein, dass sie den Upgradepfad ermitteln und ggf. nötige Migrationsschritte automatisch ausführen.
+
+Hierfür können in allen Upgrade-Skripten die vorherige (`FROM`) und neue (`TO`) Dogu-Version ausgelesen werden:
+
+```bash
+FROM_VERSION="${1}"
+TO_VERSION="${2}"
+```
+
+### pre-upgrade.sh - Führt alle Aktionen vor dem Upgrade des Dogus durch
+
+Das Pre-Upgrade-Skript kann als [Exposed Command](../core/compendium_de.md#exposedcommands) in der dogu.json eines Dogus definiert werden:
+
+```json
+  "ExposedCommands": [
+    {
+      "Name": "pre-upgrade",
+      "Command": "/pre-upgrade.sh"
+    }
+  ]
```

Comment:
I suggest to hint the dogu developer that there is more to the `dogu.json` than this snippet, like so:

```suggestion
{
   ...,
  "ExposedCommands": [
    {
      "Name": "pre-upgrade",
      "Command": "/pre-upgrade.sh"
    }
  ],
  ...
  }
```